### ISSUE TITLE
Enable nested fields support

### ISSUE THREAD

**Issue by BabyGeek** [typesense/typesense-dart#139]
### Is your feature request related to a problem? Please describe.
We can't filter based on nested fields

### Describe the solution you'd like
Enable the new feature "enable_nested_fields_ on schema (https://typesense.org/docs/guide/tips-for-searching-common-types-of-data.html#dates-times) working from Typesense 0.24

### Describe alternatives you've considered
Flatten schema, but in the long run, I think that typesets updated so the package should be updated

### Additional context
Just need to add a boolean and map it, then add the object type to fields

**Comment by BabyGeek** [comment]
Schema model updated

```Dart
part of models;

abstract class BaseSchema {
  /// [fields] used for querying, filtering and faceting.
  final Set<Field> fields;

  BaseSchema(this.fields);

  Map<String, dynamic> toMap() {
    final map = <String, dynamic>{};
    map['fields'] = fields.map((field) => field.toMap()).toList();
    return map;
  }

  @override
  String toString() => toMap().toString();
}

class Schema extends BaseSchema {
  /// [name] of the collection.
  final String name;

  /// Number of documents currently in the collection [name].
  final int? documentCount;

  /// A field in [fields] which will determine the order in which the search
  /// results are ranked when a `sort_by` clause is not provided during
  /// searching.
  final Field? defaultSortingField;

  /// Boolean to enable nested fields on the schema, only available for typesense 0.24 or more
  final bool? enableNestedFields;

  Schema(this.name, super.fields,
      {this.defaultSortingField, this.documentCount, this.enableNestedFields});

  factory Schema.fromMap(Map<String, dynamic> map) {
    if ((map['name'] as String).isEmpty) {
      throw ArgumentError('Ensure Schema.name is not empty');
    }
    final Set<Field> fields = (map['fields'] != null)
        ? (map['fields'] as List).map((field) => Field.fromMap(field)).toSet()
        : {};

    final String? sortingFieldName = map['default_sorting_field'];
    final Field? defaultSortingField = (fields.isNotEmpty &&
            sortingFieldName != null &&
            sortingFieldName.isNotEmpty)
        ? fields.firstWhere(
            (field) => map['default_sorting_field'] == field.name,
            orElse: () => throw _defaultSortingFieldNotInSchema(
              sortingFieldName,
            ),
          )
        : null;
    final bool? enableNestedFields = map["enable_nested_fields"];

    return Schema(map['name'], fields,
        documentCount: map['num_documents'] ?? 0,
        defaultSortingField: defaultSortingField,
        enableNestedFields: enableNestedFields);
  }

  @override
  Map<String, dynamic> toMap() {
    final map = super.toMap();
    map['name'] = name;
    if (defaultSortingField != null) {
      map['default_sorting_field'] = defaultSortingField!.name;
    }
    if (documentCount != null) {
      map['num_documents'] = documentCount;
    }
    if (enableNestedFields != null) {
      map['enable_nested_fields'] = enableNestedFields;
    }
    return map;
  }
}

class UpdateSchema extends BaseSchema {
  UpdateSchema(Set<UpdateField> fields) : super(fields);

  factory UpdateSchema.fromMap(Map<String, dynamic> map) {
    final Set<UpdateField> fields = (map['fields'] != null)
        ? (map['fields'] as List)
            .map((field) => UpdateField.fromMap(field))
            .toSet()
        : {};

    return UpdateSchema(
      fields,
    );
  }
}

ArgumentError _defaultSortingFieldNotInSchema(String name) =>
    ArgumentError('Ensure defaultSortingField "$name" is present in fields');
```


Type enum updated

```Dart

/// Enumerates the allowed field types.
///
/// [Type.auto] and [Type.stringify] are special field types that are used for
/// handling data sources with varying schema via automatic schema detection.
///
/// - [Type.auto] is used to let Typesense detect the type of the fields
/// automatically.
///
/// - [Type.stringify] (`string*`) is a way to store the field value (both
/// singular and multi-value/array values) as string.
///
/// [Type.geopoint] is used to index locations, filter and sort on them.
///
/// [Type.object] is used to let typesense know that the field is an object.
enum Type {
  string,
  int32,
  int64,
  float,
  bool,
  auto,
  stringify,
  geopoint,
  object
}

extension _Type on Type {
  String value(bool isMultivalued) {
    switch (this) {
      case Type.string:
      case Type.int32:
      case Type.int64:
      case Type.float:
      case Type.bool:
      case Type.geopoint:
      case Type.object:
        final description = toString(),
            indexOfDot = description.indexOf('.'),
            value = description.substring(indexOfDot + 1);

        return isMultivalued ? '$value[]' : value;

      case Type.auto:
        return 'auto';

      case Type.stringify:
        return 'string*';

      default:
        return '';
    }
  }
```

**Comment by BabyGeek** [comment]
Created a PR to enable this #140

**Comment by BabyGeek** [comment]
Fixed with #140
